ocaml/xapi/binpack.ml

```diff
@@ -171,6 +172,24 @@ let simulate_failure config dead_host =
   let placement = List.map (fun (vm, host) -> vm, (if List.mem_assoc vm plan then assoc "simulate_failure" vm plan else host)) config.placement in
   { config with hosts = hosts; placement = placement; num_failures = config.num_failures - 1 }
 
+let simulate_failure_approximation config =
+  let rec sum_of_first_n n l = match l with
+    | [] -> failwith "sum_of_first_n"
+    | x::l -> if n=1 then x else x + (sum_of_first_n (n-1) l)
+  in
+  (* Assume all VMs are as big as the biggest *)
+  let vm_size = List.fold_left (fun acc (_, size) -> max acc size) 0L config.vms in
+
+  (* How many VMs can each host support to power on, ascending order *)
+  let host_capacity = List.sort compare (List.map (fun (_, h_size) -> Int64.(to_int (div h_size vm_size))) config.hosts) in
```

Comment:
A better name would be `hosts_by_capacity` and a comment that this is in ascending order.